spec.bs

```diff
@@ -1165,6 +1369,173 @@ Note that there is no `paymentRequestOrigin` field in
+
+1. Remove any entries from |allowed_algorithms| that are not of type
+    {{PublicKeyCredential}}.
+
+    Issue: <span style="background-color: red">TODO</span>: Should we expand
+    the definition in the spec? Other types of public key credentials could
+    be allowable; however, the browserBoundPublicKey would need to take on
+    a more broad meaning. For example an identifier might need to be used
+    instead of the public key in clientData json.
+
+1. Remove any entries from |allowed_algorithms| that the user agent does
+    not support.
+
+    Note: User agents may support different sets of a BBK algorithms on
+    different device platforms.
+
+1. If |allowed_algorithms|'s [=list/size=]
+
+    <dl class="switch">
+        : is zero
+        :: Stop, returning null.
+
+            Note: The user agent does not add any browser bound outputs in this
+            case.
+
+        : is non-zero
+        :: Proceed to create the keypair:
+
+            1. Create the key pair according to available platform
+                specific APIs which are expected to provide an identifier for
+                later retrieval of the keypair from the platform. The first
+                algorithm in |allowed_algorithms| is provided to the platform
+                APIs as part of keypair creation parameters.
+
+                Note: A parameter indicating the private key is created within
+                    a secure element of the device.
+
+            1. If the platform specific step above failed for any reason,
+                return null.
+
+            1. Let |bbk| be the platform specific handle on the keypair
+                from the step above.
+
+            1. Let |bbk_id| the platform specific identifier of the keypair
+                form the step above.
+
+            1. Return the tuple (|bbk|, |bbk_id|).
+
+            Note: Some platforms may provide identifiers for keypairs, while
+                others may provide a wrapped (encrypted) private key.
+                In either case the result can be presented as a buffer, and the
+                keypair handle subsequently retrieved from platform APIs.
+
+            Note: The user agent may provide a software implemention.
+
+    </dl>
+
+Issue: The spec could specify to prefer algorithms in hardware storage (in
+    given order) then algorithms in software (in the same given order). For the
+    time being this topic may be moot since Chrome plans to support only 1
+    algorithm per platform.
+
+## Binding a KeyPair Internal Method ## {#sctn-binding-a-keypair}
+
+<dfn for="BrowserBoundKeyStore" method>BindKeyPair(bbk_id, credential_id)</dfn>:
+
+Given the |bbk_id|, the platform keypair identifier, and |credential_id|, the SPC credential_id (i.e. passkey id), stores the identifiers in the |browser_bound_map|.
+
+1. Insert the browser bound key id into the map. |browser_bound_map|[|credential_id|] = |bbk_id|.
+
+1. Serialize the map to disk.
```

Comment:
Again avoid talking about how the map is stored.

(But talking about what to do in the algorithm if it fails is good. We can be generic, just like "if storing in the map fails, do X")